shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-sqlserver/src/main/antlr4/imports/sqlserver/DDLStatement.g4

```diff
@@ -91,6 +91,18 @@ columnConstraint
     : (CONSTRAINT ignoredIdentifier)? (primaryKeyConstraint | columnForeignKeyConstraint | checkConstraint)
     ;
 
+computedColumnConstraint
+    : (CONSTRAINT ignoredIdentifier)? (primaryKeyConstraint | computedColumnForeignKeyConstraint | checkConstraint)
+    ;
+
+computedColumnForeignKeyConstraint
+    : (FOREIGN KEY)? tableName (LP_ columnName RP_)? computedColumnForeignKeyOnAction*
+    ;
+
+computedColumnForeignKeyOnAction
+    : ON DELETE (NO ACTION | CASCADE) | ON UPDATE NO ACTION | NOT FOR REPLICATION
+    ;
+
```

Comment:
@Icesource Maybe we can refactor it，`fileTableClause` rule has `as`, you can give proposal and discuss together .

Reply:
@jingshanglu  ```fileTableClause``` is different from ```<computed_column_definition>```.  ```As``` in ```fileTableClause``` and ```computedColumnDefinition``` may appear at the same time.  So I think their parsing rules should be separated. Is that right?
```
CREATE TABLE
    { database_name.schema_name.table_name | schema_name.table_name | table_name }
    [ AS FileTable ]
    ( {   <column_definition>
        | **<computed_column_definition>**
        | <column_set_definition>
        | [ <table_constraint> ] [ ,... n ]
        | [ <table_index> ] }
          [ ,...n ]
          [ PERIOD FOR SYSTEM_TIME ( system_start_time_column_name
             , system_end_time_column_name ) ]
      )
    [ ON { partition_scheme_name ( partition_column_name )
           | filegroup
           | "default" } ]
    [ TEXTIMAGE_ON { filegroup | "default" } ]
    [ FILESTREAM_ON { partition_scheme_name
           | filegroup
           | "default" } ]
    [ WITH ( <table_option> [ ,...n ] ) ]
[ ; ]
```

Reply:
@Icesource  It is possible that the 'columnConstraint' has the same effect as the 'computedColumnConstraint' you defined.We can eliminate some of the redundant code.

Reply:
@jingshanglu Of course, I try to extract the redundant code into a syntax ```commonColumnConstraint``` in the new commit. Do you think this change is reasonable? 

Besides, I add the syntax ```constraintName``` after ```indexName``` in baseRule.g4 because [ CONSTRAINT constraint_name ] appeared many times.

Reply:
@jingshanglu I try to extract the same code in ```columnConstraint``` and ```computedColumnConstraint``` into a new syntax. But if I do this, columnConstraintContext will change so I need to change SQLVistor too. This may not be a good idea. 
```columnConstraint``` and ```computedColumnConstraint``` have similar syntax but different scenarios. I don't find a good way to eliminate the same code. Do you have any suggestions?

Reply:
@jingshanglu  I think this may not be a problem, because the SQL server official website also rewrites this similar syntax twice. According to the current writing method, it can maintain higher consistency and readability with the official website grammar definition. They are not the same and the scope of action is different.